TWLight/users/models.py

```diff
@@ -256,6 +223,212 @@ def encode_wp_username(self, username):
         result = urllib.parse.quote(username)
         return result
 
+    @property
+    def wp_editcount(self):
+        """
+        Fetches latest editcount from EditorLogs related to this editor.
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        int : Most recently recorded Wikipedia editcount
+        """
+        try:
+            return (EditorLog.objects.filter(editor=self).latest("timestamp")).editcount
+        except models.ObjectDoesNotExist:
+            pass
+
+    @property
+    def wp_editcount_updated(self):
+        """
+        Fetches timestamp of latest editcount from EditorLogs related to this editor.
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        datetime.datetime : datetime that editcount was recorded
+        """
+        try:
+            return (EditorLog.objects.filter(editor=self).latest("timestamp")).timestamp
+        except models.ObjectDoesNotExist:
+            pass
+
+    def wp_editcount_prev(
+        self,
+        current_datetime: timezone = None,
+    ):
+        """
+        Fetches 30-day old editcount from EditorLogs related to this editor.
+        Parameters
+        ----------
+        self
+        current_datetime : timezone
+            optional timezone-aware timestamp override that represents now()
+
+        Returns
+        -------
+        int : 30-day old Wikipedia editcount, if available.
+        """
+        if not current_datetime:
+            current_datetime = timezone.now()
+        try:
+            return (
+                EditorLog.objects.filter(
+                    editor=self, timestamp__lte=current_datetime - timedelta(days=30)
+                ).latest("timestamp")
+            ).editcount
+        except models.ObjectDoesNotExist:
+            pass
+
+    def wp_editcount_prev_updated(
+        self,
+        current_datetime: timezone = None,
+    ):
+        """
+        Fetches timestamp of 30-day old editcount from EditorLogs related to this editor.
+        Parameters
+        ----------
+        self
+        current_datetime : timezone
+            optional timezone-aware timestamp override that represents now()
+
+        Returns
+        -------
+        datetime.datetime : datetime that editcount_prev was recorded
+        """
+        if not current_datetime:
+            current_datetime = timezone.now()
+        try:
+            return (
+                EditorLog.objects.filter(
+                    editor=self, timestamp__lte=current_datetime - timedelta(days=30)
+                ).latest("timestamp")
+            ).timestamp
+        except models.ObjectDoesNotExist:
+            pass
+
+    def wp_editcount_recent(
+        self,
+        current_datetime: timezone = None,
+    ):
+        """
+        Calculates recent editcount based on EditorLogs related to this editor.
```

Comment:
Let's add that the recent edit count is used to determine whether a user may be bundle eligible or not.

Reply:
done!